File: 프로그래머스/Lv.0/최빈값 구하기/120812.java
import java.util.*;

class Solution {
    public int solution(int[] array) {
        int answer = 0;

        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for (int num : array) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        int maxKey = -1;
        int maxValue = -1;
        boolean multiFlag = false;
        for (int key : map.keySet()) {
            int value = map.get(key);

            if (value > maxValue) {
                maxKey = key;
                maxValue = value;
                multiFlag = false;
            } else if (value == maxValue) {
                multiFlag = true;
            }
        }

        if (multiFlag) {
            answer = -1;
        } else {
            answer = maxKey;
        }

        return answer;
    }
}

File: 프로그래머스/Lv.0/최빈값 구하기/120812.js
function solution(array) {
    let answer = 0;
    let maxCnt = 0;

    const map = new Map();
    array.forEach(num => {
        const cnt = (map.get(num) || 0) + 1;

        if (cnt > maxCnt) {
            answer = num;
            maxCnt = cnt;
        } else if (cnt === maxCnt) {
            answer = -1;
        }

        map.set(num, cnt);
    });

    return answer;
}
